marklogic-client-api/src/test/java/com/marklogic/client/test/rows/FromSearchWithOptionsTest.java

```diff
@@ -0,0 +1,40 @@
+package com.marklogic.client.test.rows;
+
+import com.marklogic.client.row.RowRecord;
+import com.marklogic.client.test.junit5.RequiresML12;
+import com.marklogic.client.type.PlanSearchOptions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(RequiresML12.class)
+class FromSearchWithOptionsTest extends AbstractOpticUpdateTest {
+
+	@Test
+	void bm25() {
+		PlanSearchOptions options = op.searchOptions()
+			.withScoreMethod(PlanSearchOptions.ScoreMethod.BM25)
+			.withBm25LengthWeight(20.5);
+		List<RowRecord> rows = resultRows(
+			op.fromSearch(op.cts.wordQuery("contents"), null, null, options)
+				.offsetLimit(0, 5)
+		);
+		assertEquals(5, rows.size());
```

Comment:
Include a message here noting that we're not actually testing that BM25 scoring is being calculated correctly - we trust the server to do that. We just want to make sure that selecting BM25 and including a length/weight doesn't cause any problems. 

Ideally though, there'd be some way to detect that the length/weight is actually submitted to the server - for all we know, the Java Client may be ignoring it. To achieve that - the best approach is probably to try putting in an invalid value - maybe a negative one, or something outside the accepted bounds, if such bounds exist? - and then verifying that the server throws an error. 

If that's not possible, then just include the assertion message and merge away.